File: src/main/java/dev/sky_lock/pocketlifevehicle/task/AbstractWarning.java
package dev.sky_lock.pocketlifevehicle.task;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;

/**
 * @author sky_lock
 */

class AbstractWarning {
    private static final String WARNING = ChatColor.RED + "⚠⚠WARNING⚠⚠";
    private int count;


    int getCount() {
        return count;
    }

    void setCount(int count) {
        this.count = count;
    }

    void send(Player player, String subTitle) {
        player.sendTitle(WARNING, subTitle, 1, 20, 1);
    }

    void stop(Player player) {
        player.resetTitle();
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/task/AbstractWarning.kt
package dev.sky_lock.pocketlifevehicle.task

import dev.sky_lock.pocketlifevehicle.text.Line
import net.kyori.adventure.title.Title
import org.bukkit.entity.Player

/**
 * @author sky_lock
 */
open class AbstractWarning {
    var count = 0

    fun send(player: Player, subTitle: Line) {
        val title = Title.title(WARNING.toComponent(), subTitle.toComponent())
        player.showTitle(title)
    }

    fun stop(player: Player) {
        player.resetTitle()
    }

    companion object {
        private val WARNING = Line().red("⚠⚠WARNING⚠⚠")
    }
}